src/main/kotlin/reactor/kotlin/extra/math/MathFluxExtensions.kt

```diff
@@ -101,10 +348,72 @@ fun <T> Flux<T>.sum(mapper: (T) -> Number): Mono<Long>
  *
  * @param mapper a lambda converting values to [Number]
  * @author Simon Baslé
- * @since 3.1.1
+ * @since 1.0.0
  */
 fun <T> Flux<T>.sumDouble(mapper: (T) -> Number): Mono<Double>
         = MathFlux.sumDouble(this, Function(mapper))
+
+/**
+ * Extension to map arbitrary values in a [Flux] to [Number]s and return the sum of these
+ * Numbers as a [Mono] of [BigInteger]
+ *
+ * Note that numbers will be mapped to [BigInteger] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T> Flux<T>.sumBigInt(mapper: (T) -> Number): Mono<BigInteger>
+        = MathFlux.sumBigInteger(this, Function(mapper))
+
+/**
+ * Extension to map arbitrary values in a [Flux] to [Number]s and return the sum of these
+ * Numbers as a [Mono] of [BigInteger]
+ *
+ * Note that numbers will be mapped to [BigInteger] using Java standard conversions
+ * This may lead to arbitrary precision gain or loss if the source flux contains different types
+ * Please choose an appropriate method based on the expected types in the source flux.
+ *
+ * @param mapper a lambda converting values to [Number]
+ * @author Mark Pruden
+ * @since 1.1.5
+ */
+fun <T> Flux<T>.sumBigDecimal(mapper: (T) -> Number): Mono<BigDecimal>
+        = MathFlux.sumBigDecimal(this, Function(mapper))
+
+/**
+ * General purpose extension function to map arbitrary values in a [Flux] to [Number]s and
+ * return the average of these Numbers as a [Mono] of [Number].
+ * The resultant [Mono] will have the same [Number] type as the output of the mapping function
+ *
+ * If the result type (or precision) needs to be changed then a specific averageType() method
+ * should be used in preference to this method. e.g. averageDouble(mapper) can be used to compute
```

Comment:
as in `sumAll(mapper)`, this shouldn't reference `averageDouble(mapper)` but simply `averageDouble()`

Reply:
resolved